apps/community/src/utils/http.ts

```diff
@@ -0,0 +1,66 @@
+import type { BaseResponse } from '~/types/api';
+
+interface RequestParams {
+  method: string;
+  url: string;
+  options?: RequestInit;
+  data?: unknown;
+}
+
+async function request<Response>({
+  method,
+  url,
+  options = {},
+  data,
+}: RequestParams): Promise<BaseResponse<Response>> {
+  const config: RequestInit = {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+      ...(options.headers || {}),
+    },
+    ...options,
+  };
+
+  if (data) {
+    config.body = JSON.stringify(data);
+  }
+
+  const res = await fetch(url, config);
+  if (!res.ok) {
+    throw new Error(`Status: ${res.status}`);
+  }
+  const responseData = await res.json();
+  return responseData as BaseResponse<Response>;
+}
+
+const http = {
+  get: async function get<Response = unknown>(
+    url: string,
+    options?: RequestInit,
+  ): Promise<BaseResponse<Response>> {
+    return request<Response>({ method: 'GET', url, options });
+  },
+  post: async function post<Request, Response = unknown>(
+    url: string,
+    data?: Request,
+    options?: RequestInit,
+  ): Promise<BaseResponse<Response>> {
+    return request<Response>({ method: 'POST', url, options, data });
+  },
+  put: async function put<Request = unknown, Response = unknown>(
+    url: string,
+    data?: Request,
+    options?: RequestInit,
+  ): Promise<BaseResponse<Response>> {
+    return request<Response>({ method: 'PUT', url, options, data });
+  },
+  delete: async function deleteMethod<Response = unknown>(
+    url: string,
+    options?: RequestInit,
+  ): Promise<BaseResponse<Response>> {
+    return request<Response>({ method: 'DELETE', url, options });
+  },
```

Comment:
```suggestion
  delete: async function<Response = unknown>(
    url: string,
    options?: RequestInit,
  ): Promise<BaseResponse<Response>> {
    return request<Response>({ method: 'DELETE', url, options });
  },
```
delete 예약어로 인해서 해당 부분만 `deleteMethod`로 선언해주었네요. 익명함수를 통해서 해당 부분을 수정할 수 있을 것 같은데 어떻게 생각하시나요 ?

Reply:
동의합니다, 다른 함수도 업데이트 할게요